First Cookbook merge

willimsbw opened the pull request:
Small readme update, addition of cookbook folder and first example plugin

Commit:
Pulling cookbook folder and readme changes from Master onto 69

stefanfroelicher-agilent committed:
Merge pull request #3 from willimsbw/69-cookbook-migration

Pulling cookbook folder and readme changes from Master onto 69

stefanfroelicher-agilent commented on cookbook/readme.md:
```diff
@@ -0,0 +1,155 @@
+and the [top-level readme](../README.md) in this repo**
+
+This cookbook is intended to provide you with a series of example plugins and code samples for developing SLIMS plugins 
+utilizing the slimsgate api. It is not intended to replace the slims development manual, [Apache's camel documentation](https://camel.apache.org/components/4.0.x), 
```
Here I think we should not put 4.0.x version of camel doc -> since we are still in 3.x 
https://camel.apache.org/components/3.14.x/index.html

stefanfroelicher-agilent commented on cookbook/readme.md:
```diff
@@ -0,0 +1,155 @@
+
+   * Examples to illustrate commonly-utilized camel route designs and camel route endpoints other than "direct" routes that run some beans
+   * Concepts explored:
+     * [Netty](https://camel.apache.org/components/4.0.x/netty-component.html) (TCP connection plumbing, handled by camel)
```
here same as previous comment -> put the 3.x version of the doc (https://camel.apache.org/components/3.14.x/index.html)

stefanfroelicher-agilent reviewed (changes requested):
All good as already discussed but just the camel doc

Commit:
NO_JIRA - updated camel links to 3.14.x

willimsbw committed:
Merge pull request #4 from willimsbw/69-cookbook-migration

NO_JIRA - updated camel links to 3.14.x

stefanfroelicher-agilent reviewed (approved):
Sounds good like this

Commit:
NO_JIRA - added cookbook plugin to settings.gradle

Commit:
NO_JIRA - updated readme settings.gradle example to match updated settings.gradle

willimsbw committed:
Merge pull request #5 from willimsbw/69-cookbook-migration

Adding first cookbook plugin to settings.gradle and updating readme to match

willimsbw commented:
@stefanfroelicher-agilent just added the cookbook's first plugin to settings.gradle and readme.md settings.gradle example -- you can view the new changes.

Commit:
NO_JIRA - added additional context comments in a few files

- Updated template 'resources' files with some additional context and included the same comments in plugin's 'resources' files
- Added comment to template springconfig
- Added some comments to plugin configuration file to provide a little context on what it does
- Updated basic-crud-actions readme to link to short explainer on using ConvertRecordsService

willimsbw committed:
Fixed a typo in a hyperlink

stefanfroelicher-agilent reviewed (approved):
Perfect for me

Commit:
NO_JIRA - small clarifications and typo updates in various comments based on review

Commit:
NO_JIRA fixing final two comments

willimsbw committed:
Merge pull request #6 from willimsbw/69-cookbook-migration

NO_JIRA - added additional context comments in a few files

stefanfroelicher-agilent reviewed (approved):
it looks all good for me
